AdminWebsite/AdminWebsite/ClientApp/src/app/booking/add-participant/add-participant.component.ts

```diff
@@ -654,12 +654,20 @@ export class AddParticipantComponent extends BookingBaseComponent implements OnI
         const _linkedParticipants: LinkedParticipantModel[] = [];
         if (newParticipant.hearing_role_name.toLowerCase() === HearingRoles.INTERPRETER) {
             if (this.editMode) {
-                const linkedParticipant = newParticipant.linked_participants[0];
-                const interpretee = this.hearing.participants.find(p => p.id === linkedParticipant.linkedParticipantId);
-                interpretee.linked_participants = [];
-                linkedParticipant.linkedParticipantId = this.getInterpreteeId(newParticipant.interpreterFor);
-                linkedParticipant.participantId = newParticipant.id;
-                _linkedParticipants.push(linkedParticipant);
+                if (this.localEditMode) {
+                    const linkedParticipant = newParticipant.linked_participants[0];
```

Comment:
3 nested if-statements is a bit of a code smell. recent PRs have been failing due to the cognitive complexity growing between each PRs. Could you introduce some private methods here to break this down, please? Thank you